tests/Interface/core/ImageContainerTest.cpp

```diff
@@ -24,6 +24,45 @@
 #include <buddy/Core/Container.h>
 #include <buddy/DIP/ImageContainer.h>
 
+bool compare_flt(float a,float b)
+{
+	if(a==b) return true;
+        float eps = a-b;
+        if(eps<0)eps=-eps;
+        if(eps<FLT_EPSILON) return true;
+        return false;
+}
+template<typename T,size_t N>
+bool testcvdp(cv::Mat testImgcv,Img<T,N> testImg,bool norm = false)
+{
+	int cvn = testImgcv.dims;
+	if(cvn!=N) return false;
+	for(int i=0;i<N;i++)
+	{
+		if(testImgcv.size[i]!=testImg.getSizes()[i]) return false;
+	}
+	T *data= testImg.getData();
+	if(N==2)
+	{
+		int k=0;
```

Comment:
Do you really require signedness from the variable here? You can use unsigned in all cases where it is not required.